controllers/rest/api.go

```diff
@@ -575,56 +566,89 @@ func (p *RESTAPI) Initialize(ctxt context.Context) error {
 }
 
 //SQLConnectionFromConfig get db connection based on a Config
-func (p *RESTAPI) SQLConnectionFromConfig(config *model.DBConfig) (*sql.DB, *gorm.DB, error) {
+func (p *RESTAPI) SQLConnectionFromConfig(config *model.DBConfig) (*sql.DB, *gorm.DB, string, error) {
 	var connStr string
 	var err error
 
-	switch config.Driver {
-	case "sqlite3":
-		//check if file exists and if not create it. We only do this if a memory only db is NOT asked for
-		//(Note that if it's a combination we go ahead and create the file) https://www.sqlite.org/inmemorydb.html
-		if config.Database != ":memory:" {
-			if _, err = os.Stat(config.Database); os.IsNotExist(err) {
-				_, err = os.Create(strings.Replace(config.Database, ":memory:", "", -1))
-				if err != nil {
-					return nil, nil, model.NewError(fmt.Sprintf("error creating sqlite database '%s'", config.Database), err)
+	if config.AwsIam {
+		dbName := config.Database
+		dbUser := config.User
+		dbHost := config.Host
+		dbPort := config.Port
+		dbEndpoint := fmt.Sprintf("%s:%d", dbHost, dbPort)
+		region := config.AwsRegion
+
+		cfg, err := awsconfig.LoadDefaultConfig(context.TODO())
+		if err != nil {
+			log.Printf("aws configuration error: " + err.Error())
+		}
+
+		authenticationToken, err := auth.BuildAuthToken(
+			context.TODO(), dbEndpoint, region, dbUser, cfg.Credentials)
+		if err != nil {
+			log.Printf("failed to create aws authentication token: " + err.Error())
+		}
+
+		switch config.Driver {
+		case "mysql":
+			connStr = fmt.Sprintf("%s:%s@tcp(%s)/%s?tls=true&allowCleartextPasswords=true",
```

Comment:
you're essentially creating your own connection string for mysql but it doesn't account for the parameters that are needed to get the mysql stuff working with GORM see the mysql connection string `connStr = fmt.Sprintf("%s:%s@tcp(%s:%s)/%s?sql_mode='ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION'&parseTime=true",`